app/ui/src/app/customizations/api-connector/api-connector-detail/api-connector-detail.component.ts

```diff
@@ -34,9 +40,17 @@ export class ApiConnectorDetailComponent implements OnInit {
       .switchMap(([id, apiConnectors]: [string, ApiConnectors]) =>
         apiConnectors.filter(apiConnector => apiConnector.id == id)
       );
+
+    this.icon$ = this.apiConnectorData$.mergeMap(connector => {
+      if (connector.icon.startsWith('db:')) {
```

Comment:
Have seen this `if` statement a lot.  Maybe there's a better place to check this.  My suggestion would be to enhance the service and maybe just reset/add the icon property in there, as with this current code the person refactoring it has to go through all these files and discover/update these if statements.